backend/api/src/algolia/migration.rs

```diff
@@ -2,231 +2,135 @@ use algolia::{
     model::attribute::{Attribute, FacetAttribute, SearchableAttributes},
     request::SetSettings,
 };
-use futures::future::BoxFuture;
 
-#[derive(Copy, Clone, Debug)]
-pub enum ResyncKind {
-    None,
-    Complete,
-}
-
-fn bad_batch_object<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    _jig_index: &'a str,
-    _course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    Box::pin(async move {
-        client.delete_object(media_index, "batch").await?;
-        Ok(())
-    })
-}
-
-fn set_searchable_fields_v1<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_searchable_fields_v2`
-    empty(client, media_index, jig_index, course_index)
-}
+use sqlx::PgConnection;
 
-fn set_attributes_for_faceting_v1<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v3`
-    empty(client, media_index, jig_index, course_index)
-}
+use hashfn::hashfn;
 
-fn set_searchable_fields_v2<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_searchable_fields_v3`
-    empty(client, media_index, jig_index, course_index)
-}
-fn set_attributes_for_faceting_v2<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v3`
-    empty(client, media_index, jig_index, course_index)
-}
-
-fn set_attributes_for_faceting_v3<'a>(
-    client: &'a super::Inner,
-    media_index: &'a str,
-    jig_index: &'a str,
-    course_index: &'a str,
-) -> BoxFuture<'a, anyhow::Result<()>> {
-    // superceeded by `set_attributes_for_faceting_v4`
-    empty(client, media_index, jig_index, course_index)
-}
+const JIG_INDEX: &str = "jig_index";
+const MEDIA_INDEX: &str = "media_index";
+const COURSE_INDEX: &str = "course_index";
```

Comment:
Possible to reuse these from https://github.com/ji-devs/ji-cloud/pull/2739/files#diff-87eb355515c2b9aefffae7bab41c4c8616902b5fd2aae28effd8606c6af30dc7R35-R37 ?